contrib/gitian-build.sh

```diff
@@ -1,406 +1,389 @@
-		exit 1
-	    fi
-           ;;
+        -s | --sign)
+            if [ -n "$2" ]; then
+                SIGNER=$2
+                shift
+            else
+                echo 'Error: "--sign" requires a PGP signer.'
+                exit 1
+            fi
+            ;;
         # Operating Systems
-        -o|--os)
-	    if [ -n "$2" ]
-	    then
-		linux=false
-		windows=false
-		osx=false
-		if [[ "$2" = *"l"* ]]
-		then
-		    linux=true
-		fi
-		if [[ "$2" = *"w"* ]]
-		then
-		    windows=true
-		fi
-		if [[ "$2" = *"x"* ]]
-		then
-		    osx=true
-		fi
-		shift
-	    else
-		echo 'Error: "--os" requires an argument containing an l (for linux), w (for windows), or x (for Mac OSX)\n'
-		exit 1
-	    fi
-	    ;;
-	# Help message
-	-h|--help)
-	    echo "$usage"
-	    exit 0
-	    ;;
-	# Commit or branch
-	-c|--commit)
-	    commit=true
-	    ;;
-	# Number of Processes
-	-j)
-	    if [ -n "$2" ]
-	    then
-		proc=$2
-		shift
-	    else
-		echo 'Error: "-j" requires an argument'
-		exit 1
-	    fi
-	    ;;
-	# Memory to allocate
-	-m)
-	    if [ -n "$2" ]
-	    then
-		mem=$2
-		shift
-	    else
-		echo 'Error: "-m" requires an argument'
-		exit 1
-	    fi
-	    ;;
-	# URL
-	-u)
-	    if [ -n "$2" ]
-	    then
-		url=$2
-		shift
-	    else
-		echo 'Error: "-u" requires an argument'
-		exit 1
-	    fi
-	    ;;
-        # kvm
-        --kvm)
-            lxc=false
+        -o | --os)
+            if [ -n "$2" ]; then
+                DESCRIPTORS=()
+                SIGN_DESCRIPTORS=()
+                if [[ "$2" == *"l"* ]]; then
+                    DESCRIPTORS+=('linux')
+                fi
+                if [[ "$2" == *"w"* ]]; then
+                    DESCRIPTORS+=('win')
+                    SIGN_DESCRIPTORS+=('win-signed')
+                fi
+                if [[ "$2" == *"x"* ]]; then
+                    DESCRIPTORS+=('osx')
+                    SIGN_DESCRIPTORS+=('osx-signed')
+                fi
+                shift
+            else
+                echo 'Error: "--os" specify os: l (linux), w (windows), or x (Mac OSX)'
+                exit 1
+            fi
+            ;;
+        # Help message
+        -h | --help)
+            echo "$usage"
+            exit 0
+            ;;
+        # Commit or branch
+        -c | --commit)
+            commit=true
+            ;;
+        # Number of Processes
+        -j)
+            if [ -n "$2" ]; then
+                proc=$2
+                shift
+            else
+                echo 'Error: "-j" requires an argument'
+                exit 1
+            fi
+            ;;
+        # Memory to allocate
+        -m)
+            if [ -n "$2" ]; then
+                mem=$2
+                shift
+            else
+                echo 'Error: "-m" requires an argument'
+                exit 1
+            fi
+            ;;
+        # lxc
+        --lxc)
+            USE_LXC=1
+            ;;
+        # docker
+        --docker)
+            USE_DOCKER=1
```

Comment:
do we need to also set `USE_LCX=0` here and vice-versa on `--lxc`?

Reply:
It would be for the case where someone uses both ?

There is some error handling for this case:
```bash
# Control the selection of a single virtualisation software
if [ $(($USE_LXC + $USE_DOCKER)) -ge 2 ]; then
    echo "$scriptName: Specify a single virtualisation solution between Docker, LXC or KVM."
    exit
fi
```
It will raise an error message if both are used.

Reply:
Yeah that works. If we get a lot of reports of people having either of these in ENV outside the script we can change it later.